bcycle-backend/Controllers/TripsController.cs

```diff
@@ -1,109 +1,76 @@
-﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
+﻿using System.Collections.Generic;
 using System.Threading.Tasks;
-using bcycle_backend.Data;
 using bcycle_backend.Models;
 using bcycle_backend.Models.Dto;
+using bcycle_backend.Security;
+using bcycle_backend.Services;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using Microsoft.Extensions.Configuration;
 
 namespace bcycle_backend.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
+    [Authorize]
     public class TripsController : ControllerBase
     {
-        private readonly IConfiguration _configuration;
-        private readonly BCycleContext _dbContext;
+        private readonly TripService _tripService;
 
-        public TripsController(IConfiguration configuration, BCycleContext dbContext) : base()
+        public TripsController(TripService tripService)
         {
-            this._configuration = configuration;
-            this._dbContext = dbContext;
+            _tripService = tripService;
         }
 
-
-        // temporary testing placeholder
-        [HttpPost("fakeuser")]
-        public async Task<ActionResult> CreateFakeUser()
-        {
-            User user = await UserDataHelper.GetCurrentUserDetails();
-            _dbContext.Add(user);
-            await _dbContext.SaveChangesAsync();
-            return Ok();
-        }
-
-        // GET api/trips
+        // GET /api/trips
         [HttpGet]
         public async Task<ActionResult<ResultContainer<IEnumerable<TripDto>>>> Get()
         {
-            int userID = (await UserDataHelper.GetCurrentUserID()).Value;
-            return new ResultContainer<IEnumerable<TripDto>>(
-                await _dbContext.Trips
-                    .Where(t => t.UserID == userID)
-                    .Include(t => t.TripPhotos).Include(t => t.TripPoints)
-                    .OrderByDescending(t => t.Started)
-                    .Select(t => t.AsDto())
-                    .ToListAsync());
+            var trips = await _tripService.GetAll(User.GetId()).ToListAsync();
+            return new ResultContainer<IEnumerable<TripDto>>(trips);
         }
 
-        // GET api/trips/5
+        // GET /api/trips/{id}
         [HttpGet("{id}")]
         public async Task<ActionResult<ResultContainer<TripDto>>> Get(int id)
         {
-            return new ResultContainer<TripDto>((await _dbContext.GetMyTrip(id)).AsDto());
+            var trip = await _tripService.GetUserTripAsync(id, User.GetId());
+            if (trip == null)
```

Comment:
Please avoid this coding style, as it’s easy to introduce mistakes if you look solely at the indentation. It’s better to add `{}` and avoid this situation altogether, or put the `if` and the following statement on one line.

Reply:
I really doesnt like that c#'s style of putting start bracket in new line, beacause it take a lot of space especially for single line if statement bodies, but ok. Can do

Reply:
Done